scripts/mod_loader/modapi/board.lua

```diff
@@ -1,96 +1,691 @@
 local DequeList = require("scripts/mod_loader/deque_list")
 
-BoardClass = Board
+-- TODO: test and find out what is saved in savegame, and what is not.
 
-BoardClass.MovePawnsFromTile = function(self, loc)
-	Tests.AssertEquals("userdata", type(self), "Argument #0")
-	Tests.AssertTypePoint(loc, "Argument #1")
+TERRAIN_ICE_CRACKED = -1
+TERRAIN_MOUNTAIN_CRACKED = -2
+TERRAIN_FOREST_FIRE = -3
 
-	-- In case there are multiple pawns on the same tile
-	local pawnStack = DequeList()
-	local point = Point(-1, -1)
+RUBBLE_BUILDING = 0
+RUBBLE_MOUNTAIN = 1
 
-	while self:IsPawnSpace(loc) do
-		local pawn = self:GetPawn(loc)
-		pawnStack:pushLeft(pawn)
-		pawn:SetSpace(point)
+local BoardClass = {}
+
+BoardClass.SetFire = function(self, loc, fire)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetFire",
+		params = { self, loc, fire },
+		{ "userdata|GameBoard&", "userdata|Point", "boolean|bool" },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	if fire == nil then
+		fire = true
 	end
+	
+	CUtils.SetTileFire(self, loc, fire)
+end
+
+-- returns true if tile is a forest, even if it is on fire.
+BoardClass.IsForest = function(self, loc)
+	Tests.AssertSignature{
+		ret = "bool",
+		func = "IsForest",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	return CUtils.IsTileForest(self, loc)
+end
```

Comment:
⚠ Always returns `false` in my version of the game

Reply:
Nice catch. IsTileForest was incorrectly checking if there was a forest fire.